### PR TITLE
Accept crucible disks in instance ensure request

### PR DESCRIPTION
**Pull request by jmpesp**
Add disks to the InstanceEnsureRequest, and attach as crucible disks.
Disks that are specified in the server_context.config still get
attached.

Test by running three crucible downstairs and a propolis server, then:

```
./target/release/propolis-cli -s 127.0.0.1 -p 12400 new <your_vm_name> --crucible --crucible-target 127.0.0.1:3801 --crucible-target 127.0.0.1:3802 --crucible-target 127.0.0.1:3803
```

I'll fix the incorrect commit messages when I squash this.

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -51,6 +54,12 @@ enum Command {
         /// Memory allocated to instance (MiB)
         #[structopt(short, default_value = "1024")]
         memory: u64,
+
+        #[structopt(long)]
+        crucible: bool,
+
+        #[structopt(long)]
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply cli/src/main.rs]: I agree, done in ccdf3b80e3548b6ee2d64008209e72c376e46579

File: cli/src/main.rs [hunk 2]
```diff
@@ -51,6 +54,12 @@ enum Command {
...
+        crucible_target: Vec<SocketAddr>,
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply cli/src/main.rs]: Fixed in ccdf3b80e3548b6ee2d64008209e72c376e46579.

File: cli/src/main.rs [hunk 3]
```diff
@@ -129,10 +146,25 @@ async fn new_instance(
...
         vcpus,
     };
+
     let request = InstanceEnsureRequest {
         properties,
         // TODO: Allow specifying NICs
         nics: vec![],
+        disks: if crucible {
+            vec![
+                // XXX default
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply cli/src/main.rs]: It's not actionable, good point. Removed in ccdf3b80e3548b6ee2d64008209e72c376e46579

File: client/src/api.rs [hunk 4]
```diff
@@ -21,7 +22,9 @@ pub struct InstancePathParams {
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct InstanceEnsureRequest {
     pub properties: InstanceProperties,
+    #[serde(default)]
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply client/src/api.rs:26]: Not intentional, added in 7048d26

File: server/src/lib/server.rs [hunk 5]
```diff
@@ -244,7 +247,13 @@ async fn instance_ensure(
 
     instance
         .initialize(|machine, mctx, disp, inv| {
-            let init = MachineInitializer::new(machine, mctx, disp, inv);
+            let init = MachineInitializer::new(
+                rqctx.log.new(o!()),
```
[the remark this answers is not in the document]
Comment (jmpesp) [reply server/src/lib/server.rs]: addressed in 5dfa1a7

**Review by smklein — APPROVED** [review]

**Pull request merged**